### PR TITLE
add two variables to control WD service and NLB behaviour

### PR DESCRIPTION
**Pull request by mollonado**
k8s_svc_spec and k8s_svc_annotations

### :pencil: Description
- 

### :link: Related Issues
-

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,8 @@ For more information please refer to the main [Apiary](https://github.com/Expedi
...
+| k8s_svc_annotations                          | Custom annotations for the Waggledance Kubernetes service..                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | map(string) |                  `"service.beta.kubernetes.io/aws-load-balancer-type"     = "nlb"`                  |    no    |
```
Comment (javsanbel2) [inline README.md:35]:
```suggestion
| k8s_svc_annotations                          | Custom annotations for the Waggledance Kubernetes service.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | map(string) |                  `"service.beta.kubernetes.io/aws-load-balancer-type"     = "nlb"`                  |    no    |
```

File: k8s.tf [hunk 2]
```diff
@@ -198,8 +200,13 @@ resource "kubernetes_service" "waggle_dance" {
       port        = local.wd_port
       target_port = local.wd_port
     }
-    type                        = "LoadBalancer"
-    load_balancer_source_ranges = var.ingress_cidr
+    type                              = "LoadBalancer"
+    load_balancer_source_ranges       = var.ingress_cidr
+    external_traffic_policy           = var.k8s_svc_spec.external_traffic_policy
```
Comment (javsanbel2) [inline k8s.tf:205]: What happens if `var.k8s_svc_spec.external_traffic_policy` is not specified? It is not gonna set some attributes to null? Is it that okay?

File: version.tf [hunk 3]
```diff
@@ -5,6 +5,7 @@
  */
 
 terraform {
+  experiments = [module_variable_optional_attrs]
```
Comment (javsanbel2) [inline version.tf:8]: What does this line do?

File: README.md (hunk 1 above)
Comment (patduin) [reply README.md:35]: what about the: https://github.com/ExpediaGroup/apiary-federation/pull/147/files#diff-05b5a57c136b6ff596500bcbfdcff145ef6cddea2a0e86d184d9daa9a65a288eR137

File: k8s.tf (hunk 2 above)
Comment (mollonado) [reply k8s.tf:205]: if not specified, this means passing "null".
https://developer.hashicorp.com/terraform/language/expressions/types#null

File: version.tf (hunk 3 above)
Comment (mollonado) [reply version.tf:8]: enables the `optional` modifier. This has been added to 0.15 as experiment and made to 1.3 as stable. But is completely fine and safe to use it.

File: README.md [hunk 4]
```diff
@@ -31,6 +31,8 @@ For more information please refer to the main [Apiary](https://github.com/Expedi
...
+| k8s_svc_annotations                          | Custom annotations for the Waggledance Kubernetes service.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        | map(string) |                  `"service.beta.kubernetes.io/aws-load-balancer-type"     = "nlb"`                  |    no    |
```
Comment (patduin) [inline README.md:35]: default is:
```
    "service.beta.kubernetes.io/aws-load-balancer-internal" = "true"
    "service.beta.kubernetes.io/aws-load-balancer-type"     = "nlb"
 ```

File: k8s.tf (hunk 2 above)
Comment (javsanbel2) [reply k8s.tf:205]: If you set it to null, is it gonna take the default value?

File: README.md (hunk 4 above)
Comment (javsanbel2) [reply README.md:35]: Default is added in variables.tf

Comment (mollonado) [reply README.md]: updated.

**Review by patduin — APPROVED** [review]

File: k8s.tf (hunk 2 above)
Comment (patduin) [reply k8s.tf:205]: yeah that's what the docs say

**Review by javsanbel2 — APPROVED** [review]

**Pull request merged**